### DISCUSSION TITLE
Can't query view through REST API with auth

### DISCUSSION THREAD

**Discussion by imknight** [supabase/supabase#13102] [Questions] [answered]
I created a view, no special RLS setup

```
CREATE OR REPLACE VIEW account_view AS
SELECT
*
from
	account_user
WHERE
 user_id=auth.uid()
```
Trying to query using API with the User bearer token but no record has return, how can I check whats wrong?
If I remove the where condition, the API will work.

**Comment by GaryAustin1** [comment, marked as answer]
If you have a column user_id that is a uuid type in the account_user table then you most likely don’t really have a signed in user when you make the call.   You could try auth.uid() is null, or auth.role() = ‘anon’ to check that.

**Reply by imknight** [reply]
You are right, query with "Where auth.uid() is null "returns all results. I think my login is some where broken.
